sdl-core/src/main/scala/io/smartdatalake/statusinfo/PortUtils.scala

```diff
@@ -0,0 +1,85 @@
+package io.smartdatalake.statusinfo
+
+import io.netty.channel.unix.Errors.NativeIoException
+import io.smartdatalake.util.misc.SmartDataLakeLogger
+import org.eclipse.jetty.util.MultiException
+
+import java.net.BindException
+import scala.jdk.CollectionConverters.collectionAsScalaIterableConverter
+
+object PortUtils extends SmartDataLakeLogger {
```

Comment:
Did you copy that from somewhere? Could you add a scaladoc with a short notice where you copied it from?

Reply:
Yep, it is Derived from org.apache.spark.util.Utils#startServiceOnPort and surrounding methods. I added that as a comment

Reply:
I have already derived other code from Spark. I have put this code into a separate repository having Apache Licence 2.0: https://github.com/smart-data-lake/spark-extensions. I suggest to do the same with this class.